Mathlib/Topology/ContinuousOn.lean

```diff
@@ -1214,3 +1240,14 @@ theorem continuousWithinAt_prod_iff {f : α → β × γ} {s : Set α} {x : α}
   ⟨fun h => ⟨h.fst, h.snd⟩, fun ⟨h1, h2⟩ => h1.prod h2⟩
 
 end Pi
+
+/-- If `f` is continuous on an open set `s` and continuous at each point of another
+set `t` then `f` is continuous on `s ∪ t`. -/
+lemma ContinuousOn.union_continuousAt
+    {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y]
+    {s t : Set X} {f : X → Y} (s_op : IsOpen s)
+    (hs : ContinuousOn f s) (ht : ∀ x ∈ t, ContinuousAt f x) :
```

Comment:
This is a somewhat arbitrary lemma, because modulo the distinction between ContinuousOn/ContinuousAt, this is basically just `(∀ x ∈ s ∪ t, _) ↔ (∀ x ∈ s, _) ∧ (∀ x ∈ t, _)`. But maybe it's fine.

Reply:
For what it's worth this lemma was suggested by another reviewer, I don't have a strong opinion.